geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloFeatureCollection.scala

```diff
@@ -28,16 +22,7 @@ class AccumuloFeatureCollection(source: GeoMesaFeatureSource, query: Query)
 
   override def accepts(visitor: FeatureVisitor, progress: ProgressListener): Unit =
     visitor match {
-      case v: TubeVisitor      => v.setValue(v.tubeSelect(source, query))
-      case v: ProximityVisitor => v.setValue(v.proximitySearch(source, query))
-      case v: QueryVisitor     => v.setValue(v.query(source, query))
-      case v: StatsVisitor     => v.setValue(v.query(source, query))
-      case v: SamplingVisitor  => v.setValue(v.sample(source, query))
-      case v: KNNVisitor       => v.setValue(v.kNNSearch(source,query))
-      case v: AttributeVisitor => v.setValue(v.unique(source, query))
-      case v: RouteVisitor     => v.routeSearch(source, query)
-      case v: BinVisitor       => v.binQuery(source, query)
-      case v: ArrowVisitor     => v.arrowQuery(source, query)
+      case v: GeoMesaProcessVisitor => v.execute(source, query)
```

Comment:
As I was thinking through how having this trait would work, I was wondering how we can handle partially visitor support in HBase.  

Thoughts on that?  For instance, wiring up the StatsProcess would next should be pretty quick.